Clarify install target docs. (#929)
cylc/cylc-doc: src/reference/config/writing-platform-configs.rst

 - A set of ``hosts``.
 - A ``job runner`` (formerly a ``batch system``) where Cylc can submit a job.
 - An ``install target``, representing each distinct filesystem that Cylc might

>    need to install workflow files on.

Reviewer: Suggestion:
`An ``install target``, representing the ``$HOME`` directory used on the platform.`

What Are Install Targets?
^^^^^^^^^^^^^^^^^^^^^^^^^

> An install target is an arbitrary label that represents a file system. All

Reviewer 2: (Take or leave this suggestion)
```suggestion
An install target is an arbitrary label to identify a distinct file system. All
```

platforms that see the same filesystem should specify the same install target

> so that Cylc knows where it needs to install workflow files.

Reviewer: Suggestion:
`An install target is an arbitrary label to identify the $HOME directory. All
platforms that share the same $HOME directories should specify the same install target
so that Cylc knows whether it needs to install workflow files.`

Reviewer: or maybe:
`An install target is an arbitrary label to identify the distinct filesystem used to store the $HOME directories. All platforms that share the same $HOME directories (i.e. the same files) should specify the same install target so that Cylc knows whether it needs to install workflow files
`

If you log into one system and see the same files as on another, then these two
platforms will require the same install target in ``global.cylc`` config file.

Cylc will setup each remote install target once. During setup it will:

  - Install workflow files
  - Symlink directories
  - Copy authentication keys (to allow secure communication)

Note, if missing from the configuration, the install target will default to
the platform name. If incorrectly configured, this will cause errors in

> :ref:`RemoteInit` due to Cylc finding files there already during setup.

Reviewer: Suggestion:
`Note, if missing from the configuration, the install target will default to
the platform name (which is fine if there are no other platforms using the
same $HOME directories). However, if incorrectly configured (i.e. you use
platforms which share $HOME directories but use different install targets),
this will cause errors in
:ref:`RemoteInit` due to Cylc finding files there already during setup.`
